relay-threading/src/builder.rs

```diff
@@ -0,0 +1,136 @@
+use std::any::Any;
+use std::future::Future;
+use std::io;
+use std::sync::Arc;
+
+use crate::pool::{AsyncPool, Thread};
+use crate::pool::{CustomSpawn, DefaultSpawn, ThreadSpawn};
+
+/// [`AsyncPoolBuilder`] provides a flexible way to configure and build an [`AsyncPool`] for executing
+/// asynchronous tasks concurrently on dedicated threads.
+///
+/// This builder enables you to customize the number of threads, concurrency limits, thread naming,
+/// and panic handling strategies.
+pub struct AsyncPoolBuilder<S = DefaultSpawn> {
+    pub(crate) runtime: tokio::runtime::Handle,
+    pub(crate) thread_name: Option<Box<dyn FnMut(usize) -> String>>,
+    #[allow(clippy::type_complexity)]
```

Comment:
Instead of ignoring the warning we should probably make an alias

Reply:
I wanted, but from our chat you seemed against it. Will make one.